src/pages/level/Level.tsx

```diff
@@ -19,15 +24,59 @@ const Level: FC<LevelProps> = () => {
   })
 
   const settings = useSettings()
+  const blocklyHandle = useRef<BlocklyWorkspaceHandle | null>(null)
 
   return useParamsRequired({
     shape: { id: yup.number().required().min(1) },
     children: () => (
       <Box sx={{ display: "flex" }}>
         <Controls />
         <Box component="main" sx={{ flexGrow: 1 }}>
-          <Typography>Settings: {JSON.stringify(settings)}</Typography>
-          <Typography>Level state: {JSON.stringify(level)}</Typography>
+          {/* TODO: fix style */}
+          <PanelGroup
+            direction="horizontal"
+            style={{ height: "100vh", width: "100%" }}
+            onLayout={() => {
+              // Trigger Blockly workspace resize after layout change
+              if (blocklyHandle.current) {
+                blocklyHandle.current.resize()
+              }
```

Comment:
- no need for comment, callback is self explanatory. comments only for complex/ambiguous logic
- remove unnecessary newlines
```suggestion
              if (blocklyHandle.current) blocklyHandle.current.resize()
```